CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required(VERSION 3.26)
+project(PuppetX)
+
+set(CMAKE_CXX_STANDARD 23)
+
+set(EXECUTABLE_NAME ${CMAKE_PROJECT_NAME})
+add_executable(${EXECUTABLE_NAME})
+
+add_subdirectory(main)
+add_subdirectory(puppetlogic)
+add_subdirectory(tests)
+
+target_include_directories(${EXECUTABLE_NAME}
+    PRIVATE
+        .
```

Comment:
Same question here, why we include root?

Reply:
This is actually the same as the other one xD

Reply:
But yes you're right